src/CMakeLists.txt

```diff
@@ -30,7 +30,7 @@ MACRO(IPOPT_PLUGIN NAME)
   INSTALL(TARGETS roboptim-core-plugin-${NAME}
     DESTINATION ${PLUGINDIR})
   PKG_CONFIG_USE_DEPENDENCY(roboptim-core-plugin-${NAME} ipopt)
-  PKG_CONFIG_USE_COMPILE_DEPENDENCY(roboptim-core-plugin-${NAME} roboptim-core)
+  PKG_CONFIG_USE_DEPENDENCY(roboptim-core-plugin-${NAME} roboptim-core)
```

Comment:
Those symbols should come from the main library/executable that loads the plugin, i.e. at runtime. This is what happens on Linux/Mac OS X. According to the [dlopen doc](http://pubs.opengroup.org/onlinepubs/9699919799/):

> The mode parameter describes how dlopen() shall operate upon file with respect to the processing of relocations and the scope of visibility of the symbols provided within file. When an executable object file is brought into the address space of a process, it may contain references to symbols whose addresses are not known until the executable object file is loaded.
> 
> These references shall be relocated before the symbols can be accessed. The mode parameter governs when these relocations take place and may have the following values:
> 
> RTLD_LAZY
> Relocations shall be performed at an implementation-defined time, ranging from the time of the dlopen() call until the first reference to a given symbol occurs. Specifying RTLD_LAZY should improve performance on implementations supporting dynamic symbol binding since a process might not reference all of the symbols in an executable object file. And, for systems supporting dynamic symbol resolution for normal process execution, this behavior mimics the normal handling of process execution.
> RTLD_NOW
> All necessary relocations shall be performed when the executable object file is first loaded. This may waste some processing if relocations are performed for symbols that are never referenced. This behavior may be useful for applications that need to know that all symbols referenced during execution will be available before dlopen() returns.

This can be easily confirmed with `nm -CD roboptim-core-plugin-ipopt.so | grep "U roboptim::"` on Linux (ltdl uses `RTLD_LAZY `).

Now, before doing any change to this, I would like to know more about the differences with Windows. Can we achieve the same behavior with `dlopen` and `LoadLibrary` (called by ltdl)? A quick search seems to indicate that [this is a shortcoming](http://gnuwin32.sourceforge.net/packages/edll.htm) of MSVC... Do you confirm this?